x/poa/types/genesis.go

```diff
@@ -74,3 +78,43 @@ func validateGenesisStateValidators(validators []Validator) (err error) {
 	}
 	return
 }
+
+// validateGenesisStatePrivilegeAssignments validates privilege assignments in genesis.
+//
+// Ensures that:
+// - The operator and privilege are non-empty strings.
+// - There are no duplicated assignments.
+func validateGenesisStatePrivilegeAssignments(assignments []ValidatorPrivilegeAssignment) error {
+	if len(assignments) == 0 {
+		return nil
+	}
+
+	seen := make(map[string]bool)
```

Comment:
nit: I'd use a map[string]struct{} here instead, but it's just for the form / habits, seeing that the map is tiny, and temporary.